### PR TITLE
chore: adjust Sentry configuration and add explicit DSN

### PR DESCRIPTION
**Pull request by brandonroberts**
## Description

- Adjusts the Sentry configuration for edge and server configurations
- Increase sampling for edge and server configs
- Explicitly set `dsn` from environment variable

## Tier (staff will fill in)

- [ ] Tier 1
- [ ] Tier 2
- [ ] Tier 3
- [x] Tier 4

### CODE REVIEW

File: next.config.js [hunk 1]
```diff
@@ -135,7 +135,7 @@ module.exports = withSentryConfig(
     silent: true,
 
     org: "opensauced",
-    project: "insights",
+    project: process.env.NODE_ENV === "production" ? "app-opensauced" : "beta-app-opensauced",
```
Comment (nickytonline) [inline next.config.js:138]: Does this exclude the local development environment or is it considered beta? Just asking as traffic from local devs machines' caused a bunch of noise before.

Comment (brandonroberts) [reply next.config.js:138]: Yes, local development is excluded by default

**Review by nickytonline — APPROVED** [review]
:shipit:

**Review by jpmcb — APPROVED** [review]
🐿️ 🥡

**Pull request merged**